### DISCUSSION TITLE
How to start syslog daemon in ddev web container

### DISCUSSION THREAD

**Discussion by MurzNN** [ddev/ddev#4863] [Show and tell]
Sometimes projects write logs to the syslog, and seems by default the ddev web container doesn't have any syslog receiver.
The workaround is to install and run it manually. 

But `rsyslogd` will not work out of the box with errors like this:
```
$ sudo service rsyslog start
Starting enhanced syslogd: rsyslogdrsyslogd: imklog: cannot open kernel log (/proc/kmsg): Operation not permitted.
rsyslogd: activation of module imklog failed [v8.2102.0 try https://www.rsyslog.com/e/2145 ]
.
```
And the same is with `syslog-ng`:
```
$ sudo service syslog-ng start
Starting system logging: syslog-ngsyslog-ng: Error setting capabilities, capability management disabled; error='Operation not permitted'
```

But I've found a working workaround to start it, here are the steps:
```
sudo apt update
sudo apt install syslog-ng
sudo /usr/sbin/syslog-ng -F --no-caps
```
After this - all syslog records you can watch in the syslog file:
```
tail -f /var/log/syslog
```

**Comment by rfay** [comment]
Thanks! You could set this up with a combo of a Dockerfile and post-start hooks. I think this would be more discoverable if you put it in https://github.com/ddev/ddev-contrib or Stack Overflow.

**Comment by rfay** [comment]
If it's a common issue you might consider creating a ddev-get add-on. 

You should also be able to run syslog-ng using [web_extra_daemons](https://ddev.readthedocs.io/en/latest/users/configuration/config/#web_extra_daemons)
